### ISSUE TITLE
armhf docker container not starting on arm64 with volume

### ISSUE THREAD

**Issue by phdelodder** [fg2it/grafana-on-raspberry#31]
When I try to start the container I get the following error: 

lvl=crit msg="Failed to parse /etc/grafana/grafana.ini, open /etc/grafana/grafana.ini: permission denied%!(EXTRA []interface {}=[])"

using fg2it/grafana-armhf:v5.0.3

**Comment by fg2it** [comment]
@phdelodder Thanks for reporting. Could you provide some details : 
- hardware used
- docker version
- docker command line used
- more log lines

The title of the issue mentions 'arm64', but the docker I provide (currently) only support armhf (armv7). So that could be the reason, could you try to install the [aarch64 deb package](https://github.com/fg2it/grafana-on-raspberry/releases/tag/v5.0.3) ?

Another interesting point is the perm inside de container. Could you check that ?
```bash
$ docker run -ti --entrypoint /bin/bash fg2it/grafana-armhf:v5.0.3
[inside docker]$ ls -l /etc/grafana
total 24
-rw-r----- 1 root grafana 13660 Mar 17 02:42 grafana.ini
-rw-r----- 1 root grafana  3468 Mar 17 02:42 ldap.toml
drwxr-xr-x 4 root grafana  4096 Mar 22 19:45 provisioning
```
If permissions don't allow read try to fix that and run to confirm it is indeed the problem : 
```bash
[inside docker]$ chmod +r /etc/grafana/grafana.ini
[inside docker]$ /run.sh
```
You can also try to increase log level
```bash
[inside docker]$ /run.sh cfg:default.log.level="debug"
```
or
```bash
$ docker run -ti --rm fg2it/grafana-armhf:v5.0.3 cfg:default.log.level="debug"
```

**Comment by phdelodder** [comment]
Thanks for the followup. But I restarted the docker container and now it works. I have no more explanation for it.

**Comment by fg2it** [comment]
Good, I guess. Could you tell me the hardware and os you are using ?

**Comment by phdelodder** [comment]
I'm running rock64 with dietpi. I can reproduce it it occurs always the first time you create the container, when I stop the container and start it again the issue is gone.

**Comment by fg2it** [comment]
That is strange. What version of docker are you using ? How do you start your container ?

**Comment by phdelodder** [comment]
Docker version 17.12.1-ce, build 7390fc6
docker-compose version 1.8.0, build unknown

**Comment by fg2it** [comment]
Do you mount any volume in your container ? How do you start it ? It would help if you could check the permissions, before and after launching grafana, something along : 
```bash
$ docker run -ti --entrypoint /bin/bash fg2it/grafana-armhf:v5.0.3
[inside docker]$ ls -l /etc/grafana
[inside docker]$ /run.sh
#interrupt if needed with ctrl+c
[inside docker]$ ls -l /etc/grafana
```

**Comment by phdelodder** [comment]
yes I mounted the a custom volume, without the volume it works like a charm

**Comment by fg2it** [comment]
[TL;DR]
- I invite you to provide more diagnostics on what you are facing
- my build isn't responsible (except to have chosen to follox the official one)
- we can expect to have this solve soon

[long story]
So custom volume is very likely the reason. That is why I was asking for command line in my first post...
There are some issues with the current handling of permissions with the official images (with I follow). Depending on how you do things, it could have been fixed by c8bbfc79f6c800540728926440a93bd8e2f4312f that tracks grafana/grafana-docker@291c817 (and I would have think so) or still be open (that seems to be the case since fg2it/grafana-armhf:v5.0.3 includes this change; more diagnostics would help understanding what is happening and ask Grafana staff to fix).
The official docker is expected to change with v5.1.0 release thanks to grafana/grafana-docker#146, that should solve these kinds of problem.
